src/main/java/uk/co/ramp/Outbreak.java

```diff
@@ -63,171 +70,71 @@ public void setContactRecords(Map<Integer, List<ContactRecord>> contactRecords)
+            if (activeCases == 0 && lastContact < time) {
+                randomInfectionRate = 0d;
             }
-        }
-    }
-
-    void updatePopulationState(int time, double randomInfectionRate) {
-
-        Set<Integer> alerts = new HashSet<>();
-
-        for (Case p : population.values()) {
-            EvaluateCase e = new EvaluateCase(p, diseaseProperties, distributionSampler);
-            alerts.addAll(e.checkActionsAtTimestep(time));
-
-            if (p.status() == SUSCEPTIBLE && randomInfectionRate > 0d && time > 0) {
-                boolean var = distributionSampler.uniformBetweenZeroAndOne() <= randomInfectionRate;
-                if (var) e.randomExposure(time);
-            }
-        }
-
-        if (!alerts.isEmpty()) alertPopulation(alerts, time);
-    }
-
-    void alertPopulation(Set<Integer> alerts, int time) {
 
-        for (Integer id : alerts) {
-            Case potentialInfected = population.get(id);
-            if (potentialInfected.alertStatus() == NONE && potentialInfected.status() != DEAD) {
-                potentialInfected.setNextAlertStatusChange(time + 1);
-                LOGGER.trace("ALERTED: {} - {} - {} ", id, potentialInfected.status(), potentialInfected.alertStatus());
+            if (activeCases == 0 && randomInfectionRate == 0d) {
+                LOGGER.info("There are no active cases and the random infection rate is zero.");
+                LOGGER.info("Exiting as solution is stable.");
+                break;
             }
         }
-
     }
 
-
-    void evaluateExposures(ContactRecord c, int time) {
-        Case personA = utils.getMostSevere(population.get(c.to()), population.get(c.from()));
-        Case personB = personA == population.get(c.to()) ? population.get(c.from()) : population.get(c.to());
-
-        personA.addContact(c);
-        personB.addContact(c);
-
-        boolean dangerMix = personA.isInfectious() && personB.status() == SUSCEPTIBLE;
-
-        if (dangerMix && distributionSampler.uniformBetweenZeroAndOne() < c.weight() / diseaseProperties.exposureTuning()) {
-            LOGGER.trace("       DANGER MIX");
-            EvaluateCase e = new EvaluateCase(personB, diseaseProperties, distributionSampler);
-            e.updateVirusStatus(EXPOSED, time, personA.id());
-        }
-    }
-
-    int calculateDailyStatistics(int time) {
+    void updateLogActiveCases(int time) {
+        int previousActiveCases = activeCases;
         Map<VirusStatus, Integer> stats = utils.getCmptCounts(population);
-        int activeCases = stats.get(EXPOSED) + stats.get(EXPOSED_2) + stats.get(INFECTED) + stats.get(INFECTED_SYMP);
+        activeCases = stats.get(EXPOSED) + stats.get(ASYMPTOMATIC) + stats.get(PRESYMPTOMATIC) + stats.get(SYMPTOMATIC) + stats.get(SEVERELY_SYMPTOMATIC);
 
-        CmptRecord cmptRecord = outputLog.log(time, stats);
+        CmptRecord cmptRecord = outputLog.log(time, stats, activeCases - previousActiveCases);
```

Comment:
I would push storing the state of activeCases into the implementation of LogDailyOutput. This would mean we can simply pass in (time, stats) as we were doing before, and let the LogDailyOutput class keep track of the previous cases.

Reply:
Yep that makes sense, although it does mean we have active cases being calculated in two places, but I don't think that matters.

Reply:
Nice. Yes you're right. Seems like a premature optimisation to me to calculate once and pass value through. My thinking was that this suggestion would simplify the code from a readability perspective